### ISSUE TITLE
Consultations - Attachment Log

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#1313]
* As a Ministry Coordinator
* I want an attachment log for my consultations
* so that I can upload reference material to ongoing and completed consultations

**Assumptions & Scope**	
This covers the need for Ministry coordinator's to have an attachment log for their consultations that they track.

Attachment log functionality should be the same as with FOI Requests - coordinators should be able to upload, rename, download and delete files they upload to the log. The attachment log should also look the same as you would see for a FOI request. 

Since state transitions will be managed entirely by the Ministry coordinator, there will be no enforced state transition uploads. 

What is IN scope?
Navigating too, and viewing the attachment log
Basic attachment log functionality

**Acceptance Criteria**	

**Scenario 1: Left Hand Nav Panel View**
* GIVEN you are on the review request view of a Consultation
* WHEN the left side navigation panel appears
* THEN you will see an option for "Attachments"
* AND if there are any attachments uploaded to a request
* THEN the number of attachments will appear in parentheses (ie: Attachments (2) )

**Scenario 2: Navigate to Attachment Page**
* GIVEN you are on the review request view of a request
* WHEN you click on the "attachments" panel of the left side navigation panel
* THEN you will be directed to the attachment page https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727

**Scenario 3: Attachment Page View**
* GIVEN you have clicked on the attachments panel on the left side navigation panel
* WHEN the attachments log populates
* THEN any attachments uploaded to the file will appear in an ordered list, sorted by newest uploads first
* AND each entry will include: the user that uploaded the file, the date/time the file was uploaded, and the title of the attachment
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727

**Scenario 4: Add an Attachment**
* GIVEN you are on the attachment page of a request
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* WHEN you click on the "Add Attachment" button
* THEN the upload attachment modal will Open
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727

**Scenario 5: Cancel back to Attachment Page**
* GIVEN you are on the add attachment modal from the Attachment Page
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* WHEN you click on the cancel button
* THEN the modal will close and you will remain on the attachments page
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727

**Scenario 6: Drag and Drop to add files**
* GIVEN you are on the add attachment modal from the Attachment Page
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* WHEN you drag and drop a file that is less than 500 mb into the "drag and drop" space
* THEN the file name will appear when it is uploaded, and the add attachments button will be active

**Scenario 7: Add files using the Add Files Button**
* GIVEN you are on the add attachment modal from the Attachment Page
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* WHEN you click on the "add file" button
* THEN the user will be able to select a file from their local machine to add to the upload file modal
* AND when the user selects a file and clicks "upload"
* THEN the file will appear and the "Add attachments" button will be active

**Scenario 8: Clicking On Add Attachments Button**
* GIVEN you are on the add attachment modal from the attachment page
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727
* WHEN the "Add Attachments" button is active
* AND the user has clicked on the button
* THEN the modal will close, and the attachment will be added to the attachment log in the manner described in scenario 3

**Scenario 9: Clicking On Cancel Button when Attachments uploaded**
* GIVEN you are on the add attachment modal from the attachment page
* WHEN the "Add Attachments" button is active
* AND the user has clicked the cancel button
* OR closes the modal in some other way
* THEN a pop up will ask the user "Are you sure you want to cancel (x) number of attachments will NOT be uploaded"
* AND the options will be "Yes" and "Cancel"

**Scenario 10: Cancel back to Attachment Upload**
* GIVEN the user has attempted to close the modal with an active attachment, and the pop up has prompted the user to continue
* WHEN the user clicks on cancel
* THEN the window will close and they will remain on the upload attachment page, with the attachment still present

**Scenario 11: “Continue” Out of Attachment Upload**
* GIVEN the user has attempted to close the modal with an active attachment, and the pop up has prompted the user to continue
* WHEN the user clicks on Continue
* THEN the window and attachment upload window will close and the uploaded attachment will not be uploaded to the log

**Scenario 12: Ellipsis Options On Attachment Log for Files Added**
* GIVEN a file is in the attachment log of a consultation
* WHEN a user clicks on the ellipsis of the file on the log
* THEN the user will see options to: Download, Rename, Delete
https://jacklyn808742.invisionapp.com/console/share/3W2O64AP5C/817813727

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by lmullane** [comment]
@m-prodan, this functionality is already all covered in the generic attachment log. 

How about we close this story and cover off finding consult attachments with badges?

**Comment by m-prodan** [comment]
Yes exactly - I didn't think this would have much effort since we would just re-use the component and have it specific for consults.
